person/src/homeworks/person/core/model/BasePerson.java

```diff
@@ -1,55 +1,110 @@
 package homeworks.person.core.model;
 
+import homeworks.person.core.exceptions.InvalidAgeException;
+import homeworks.person.core.exceptions.InvalidGenderException;
+import homeworks.person.core.exceptions.InvalidNameException;
 import homeworks.person.core.service.PersonAction;
+import homeworks.person.utils.Validate;
 
-public class BasePerson implements PersonAction {
+public abstract class BasePerson implements PersonAction {
     protected String name;
-    protected String designation;
+    protected String lastName;
+    protected String surname;
+    protected String nickname;
+    protected int age;
+    protected String email;
+    protected String  gender;
 
-    public BasePerson(String name, String designation)
+    protected PersonAction actions;
+
+    public BasePerson() {}
+
+
+    public void setName(String name)
     {
-        this.name = name;
-        this.designation = designation;
-    }
-    /**
-     * Set the value of name
-     * @param name the new value of name
-     */
-    public void setName(String name) {
+        if(!Validate.name(name, "^[A-Z][a-z]*$"))
+            throw new InvalidNameException();
+
         this.name = name;
     }
 
-    /**
-     * Get the value of name
-     * @return the value of name
-     */
-    public String getName() {
+    public String getName()
+    {
         return name;
     }
 
-    /**
-     * Set the value of designation
-     * @param designation the new value of designation
-     */
-    public void setDesignation(String designation) {
-        this.designation = designation;
+    public void setLastName(String lastName)
+    {
+        if(!Validate.name(lastName, "^[A-Z][a-z]*$"))
```

Comment:
@tatevikb you can make all regex as static final fields

Reply:
Done.